source/auth/tests/mongodb-oidc.rst

```diff
@@ -97,53 +112,110 @@ verify usage.  Unless otherwise specified, the tests MUST be performed with
-#. Validate the request callback inputs, including the timeout parameter if
-possible.
-#. Give a callback response with a valid accessToken and an expiresInSeconds
-that is within one minute.
-#. Ensure that a ``find`` operation adds credentials to the cache.
-#. Create a new client with the same request callback and a refresh callback.
-#. Ensure that a ``find`` operation results in a call to the refresh callback.
-#. Validate the refresh callback inputs, including the timeout parameter if
-possible.
-
-#. Ensure there is a cache with credentials that will expire in less than 5 minutes, using a client with an appropriate request callback.
-#. Create a new client with the a request callback but no refresh callback.
-#. Ensure that a ``find`` operation results in a call to the request callback.
-
-If the driver does not supports using callback hashes as part of the cache key,
-skip the next test.
-
-# Create a new client with a different request callback.
-# Ensure that a ``find`` operation adds a new entry to the cache.
-
-#. Clear the cache.
-#. Create a new client with a valid request callback that gives credentials that expire within 5 minutes and a refresh callback that gives invalid
-credentials.
-# Ensure that a ``find`` operation adds a new entry to the cache.
-#. Ensure that a subsequent ``find`` operation results in an error.
-#. Ensure that the cache has been cleared.
-
-#. Clear the cache.
-#. Create a new client using the AWS device workflow.
-#. Ensure that a ``find`` operation does not add credentials to the cache.
+Cache with refresh
+~~~~~~~~~~~~~~~~~~
+- Clear the cache.
+- Create a new client with a request callback and a refresh callback.  Both callbacks will read the contents of the ``AWS_WEB_IDENTITY_TOKEN_FILE`` location to obtain a valid access token.
+- Validate the request callback inputs, including the timeout parameter if
+  possible.
+- Give a callback response with a valid accessToken and an expiresInSeconds
+  that is within one minute.
+- Ensure that a ``find`` operation adds credentials to the cache.
+- Create a new client with the same request callback and a refresh callback.
+- Ensure that a ``find`` operation results in a call to the refresh callback.
+- Validate the refresh callback inputs, including the timeout parameter if
+  possible.
+- Close the client.
+
+Cache with no refresh
+~~~~~~~~~~~~~~~~~~~~~
+- Clear the cache.
+- Ensure there is a cache with credentials that will expire in less than 5 minutes, using a client with an appropriate request callback.
+- Create a new client with the a request callback but no refresh callback.
+- Ensure that a ``find`` operation results in a call to the request callback.
+- Close the client.
+
+Cache key includes callback
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If the driver does not support using callback hashes as part of the cache key,
```

Comment:
Could the "callback hashes" term be clarified? It is not mentioned in the spec. Some drivers are choosing to compare callbacks by reference - is that variation addressed?

Reply:
I added that option